index.js

```diff
@@ -57,9 +64,11 @@ function KinesisStream (params) {
   this.hasPriority = this.buffer.isPrioritaryMsg || this.buffer.hasPriority;
 
   // increase the timeout to get credentials from the EC2 Metadata Service
-  AWS.config.credentials = new AWS.EC2MetadataCredentials({
-    httpOptions: { timeout: 5000 }
-  });
+  if (!isLambda()) {
+    AWS.config.credentials = new AWS.EC2MetadataCredentials({
```

Comment:
What happens in the case of a lambda? does it have credentials configured elsewhere?

Reply:
yeah! credentials are in the scope of the lambda execution, set by the role configured in AWS lambda, same as EC2, but here we're forcing the cred chain to change to EC2.